android/app/src/main/java/org/jitsi/meet/AudioModeModule.java

```diff
@@ -0,0 +1,440 @@
+     */
+    private int currentMode = -1;
+
+    /**
+     * {@link AudioManager} instance used to interact with the Android audio subsystem.
+     */
+    private AudioManager audioManager;
+
+    /**
+     * {@link Handler} for running all operations on the main thread.
+     */
+    private Handler mainThreadHandler;
+
+    /**
+     * {@link Runnable} for running update operation on the main thread.
+     */
+    private Runnable mainThreadRunner;
+
+    /**
+     * {@link BluetoothManager} for detecting Bluetooth device changes in old (< M)
+     * Android versions.
+     */
+    private BluetoothManager bluetoothManager;
+
+    /**
+     * Initializes a new module instance. There shall be a single instance of this module throughout
+     * the lifetime of the application.
+     *
+     * @param reactContext the {@link ReactApplicationContext} where this module is created.
+     */
+    public AudioModeModule(ReactApplicationContext reactContext) {
+        super(reactContext);
+
+        audioManager = ((AudioManager) reactContext.getSystemService(Context.AUDIO_SERVICE));
+        mainThreadHandler = new Handler(Looper.getMainLooper());
+        mainThreadRunner = new Runnable() {
+            @Override
+            public void run() {
+                if (currentMode != -1) {
+                    updateAudioRoute(currentMode);
+                }
+            }
+        };
+        bluetoothManager = new BluetoothManager();
+
+        // Setup runtime device change detection
+        setupAudioRouteChangeDetection();
+    }
+
+    /**
+     * Gets the name for this module, to be used in the React Native bridge.
+     *
+     * @return a string with the module name.
+     */
+    @Override
+    public String getName() {
+        return MODULE_NAME;
+    }
+
+    /**
+     * Gets a mapping with the constants this module is exporting.
+     *
+     * @return a {@link Map} mapping the constants to be exported with their values.
+     */
+    @Override
+    public Map<String, Object> getConstants() {
+        final Map<String, Object> constants = new HashMap<>();
+        constants.put("DEFAULT", DEFAULT);
+        constants.put("AUDIO_CALL", AUDIO_CALL);
+        constants.put("VIDEO_CALL", VIDEO_CALL);
+        return constants;
+    }
+
+    /**
+     * Updates the audio route for the given mode.
+     *
+     * @param mode the audio mode to be used when computing the audio route.
+     * @return true if the audio route was updated successfully, false otherwise.
+     */
+    private boolean updateAudioRoute(int mode) {
+        Log.d(TAG, "Update audio route for mode: " + mode);
+
+        if (mode != DEFAULT && mode != AUDIO_CALL && mode != VIDEO_CALL) {
+            Log.d(TAG, "Invalid audio mode: " + mode);
+            return false;
+        }
+
+        if (mode == DEFAULT) {
+            audioManager.setMode(AudioManager.MODE_NORMAL);
+            audioManager.abandonAudioFocus(null);
+            audioManager.stopBluetoothSco();
```

Comment:
`audioManager.stopBluetoothSco()` is performed by `setBluetoothAudioRoute(false)` bellow. Is it necessary to call the former here as well?